bridges/modules/xcm-bridge-hub-router/src/lib.rs

```diff
@@ -130,15 +125,12 @@ pub mod pallet {
 				return T::WeightInfo::on_initialize_when_congested();
 			}
 
+			let previous_factor = Self::get_fee_factor(());
 			// if we can't decrease the delivery fee factor anymore, we don't change anything
-			if bridge.delivery_fee_factor == MINIMAL_DELIVERY_FEE_FACTOR {
+			if !Self::do_decrease_fee_factor(&mut bridge.delivery_fee_factor) {
```

Comment:
Well, actually, I didn't [want to touch](https://github.com/paritytech/polkadot-sdk/issues/8462) `pallet-xcm-bridge-hub-router`, because we will deprecate and remove it as soon as we get merged `pallet-xcm-bridge-router` here https://github.com/paritytech/polkadot-sdk/pull/6231

Reply:
> I modified `pallet-xcm-bridge-hub-router` as well because this PR is small and I would like to merge it before #6231 . After that, these changes will need to be merged in #6231 . I can do that if you want. It might be a bit of throwaway work, but it's almost insignificant.

Yes, but we will loose possibility to backport wherever we need to, because this is major change, right?

Reply:
Hmm yes, I think it's major. But being a small change, I think we can merge this one before the 2506 cutoff. Do you plan to backport #6231 also to older releases ?

Reply:
yes, we want to use it on fellowship runtimes depending on 2503 so we will "backport" to 2503

Reply:
Ok, I see. But even so, the changes to the `FeeTracker` are major. So we won't be able to use this in #6231 . We will need to create a separate PR (which will not be backported) on top of #6231 which will use the new `FeeTracker` in `pallet-xcm-bridge-router`. But we can still leave the changes to `pallet-xcm-bridge-router` in the current PR right ?

Reply:
> Ok, I see. But even so, the changes to the `FeeTracker` are major. So we won't be able to use this in #6231 . We will need to create a separate PR (which will not be backported) on top of #6231 which will use the new `FeeTracker` in `pallet-xcm-bridge-router`. But we can still leave the changes to `pallet-xcm-bridge-router` in the current PR right ?


> But we can still leave the changes to `pallet-xcm-bridge-router` in the current PR right ? 

Did you mean `pallet-xcm-bridge-hub-router` (the old one)? :)
If so, then yes, I think we can even go ahead and merge this PR, since it should be orthogonal to #6231. Once we merge/backport #6231 where needed, then we can also merge the `FeeTracker` for `pallet-xcm-bridge-router` (the new one) from #6231.

Reply:
> Did you mean pallet-xcm-bridge-hub-router (the old one)? :)

Yes, sorry, I meant `pallet-xcm-bridge-hub-router`